### PR TITLE
Restructure score article setup to allow for ScoreV1/ScoreV2 split

### PR DESCRIPTION
**Pull request by bdach**
This is a preparatory moving-around PR that is required so that I can start dealing with the fact-checks requested in https://github.com/ppy/osu-wiki/issues/6431. (Can't believe I've been roped into this.)

Attempts to follow structure suggested by @Walavouchey [on discord](https://discord.com/channels/188630481301012481/218677502141399041/987441658075447357).

Of note:

- Existing articles under `Gameplay/Score` are moved to `Gameplay/Score/ScoreV1`
- The root `Gameplay/Score` has been retouched, and gained a "ScoreV1" section
- The old `Gameplay/Score/ScoreV1` article has been nuked (as it duplicates the content from the newly-moved `Gameplay/Score/ScoreV1` subarticles), and the references to it now redirect to the added "ScoreV1" in `Gameplay/Score` section

I probably did something wrong. Oh well. Buyer beware.

## Self-check

- [x] The changes are tested against the [contribution checklist](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#self-check)

### CODE REVIEW

File: wiki/Gameplay/Score/en.md [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
+This implies a departure from the original scoring values of each individual hit object, and a change to a system that is more centred around proportions and scaling to the 1 million cap.
 
-ScoreV2 attempts to standardise all the game modes' scoring system to 1,000,000 (1 million) score at 1.00x score modifier with extra score gains from spinner's bonus for osu!, drumrolls for osu!taiko, and bananas as usual for osu!catch.
-Each of the hit objects are now part of the 1 million score and scaled accordingly rather than their own scoring values and bonus formulae.
+Aside from improved standardisation, ScoreV2 is also a workaround for an [integer overflow](https://en.wikipedia.org/wiki/Integer_overflow) issue that can arise on long and combo-intensive maps.
+Because the total score is stored as a 32-bit integer and ScoreV1 is a theoretically-unlimited scoring system, exceeding the maximum representable 32-bit integer score value of 2,147,483,647 points would cause the score counter to wrap around to negative values.
```
Comment (TicClick) [inline wiki/Gameplay/Score/en.md:31]: I think you meant to include this into the `ScoreV1` section instead

**Review by TicClick — CHANGES_REQUESTED** [review]
you also need to recreate `wiki/Gameplay/Score/ScoreV1/en.md` and make it an [index article](https://osu.ppy.sh/wiki/en/Article_styling_criteria/Formatting#index-articles). even something as simple as [`Gameplay`](https://osu.ppy.sh/wiki/en/Gameplay)-like structure would work

File: wiki/Gameplay/Score/ScoreV1/en.md [hunk 2]
```diff
@@ -1,173 +0,0 @@
...
-For yellow long notes, 300 is given per hit in the small one while 600 per hit is given to the large one.
-
-For shaker notes, each shake gives 300 and finishing the shaker gives score twice as much as a GREAT of the current combo.
-
-A typical value of *n* (4.5-5 star difficulties in old 5 star ratings system) is 80, which gives a maximum score of 1,100/2,200 at 100 combo and later combos.
-For 4 - 4.5 star difficulties, *n* is equal to 64 in which the maximum score per hit is 940/1,880.
-In the hardest case, *n* is equal to 96 in which the maximum score per hit is 1,260/2,520.
-There are lower value of *n* for even easier difficulties.
-
-Unlike [osu!d](/wiki/Game_mode/osu!)/[osu!catch](/wiki/Game_mode/osu!catch), a miss in osu!taiko will not cause a *drastic* score difference from the maximum possible score (in osu!/osu!catch, score deviation damage from a miss grows *wider* the higher the maximum combo and especially breaking in around half combo of the maximum combo).
```
Comment (TicClick) [inline wiki/Gameplay/Score/ScoreV1/en.md:73]: `osu!d`, wonder for how long this has been hanging here unnoticed

File: wiki/Gameplay/Score/en.md (hunk 1 above)
Comment (Walavouchey) [reply wiki/Gameplay/Score/en.md]: from the wording, it seems pretty intentional. dunno could go in either section, but i think it works alright as is

Comment (TicClick) [reply wiki/Gameplay/Score/en.md]: this is because I accidentally skipped over the line 30. if I consider it, yeah, it can stay where it is

**Review by Walavouchey — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**